### ISSUE TITLE
When testing a cycle event, calling application_impl::stop_offer_event is invalid.

### ISSUE THREAD

**Issue by hjkcst** [COVESA/vsomeip#666]
### vSomeip Version

v3.4

### Boost Version

1.7

### Environment

Embedded linux,ubuntu1808

### Describe the bug

When testing a cycle event, calling application_impl::stop_offer_event is invalid. The reference count for the event was not actually cleared, causing the cycle event to still notify.

The root cause is here.
file path:implementation\routing\src\event.cpp
![cycle_event](https://github.com/COVESA/vsomeip/assets/86393732/242a665f-8595-4afc-b59e-cd13ab50abea)

This problem can be solved with a few modifications.
![cycle_event_change](https://github.com/COVESA/vsomeip/assets/86393732/f10438e0-a94f-43ec-bc3a-cd97b8f6fe1d)

### Reproduction Steps

server:
0.Register handler(state、message)
1.offer_service
2.offer_event (set sycle event)
3.notify 
4.stop_offer_event
result:
The server is still notifying, and the client can still receive the notification message.

client:
0.Register handler(state、massage、availability)
1.request_service
2.request_event
3.subscribe(sycle event)
result:
It can receive a notification and continue to receive notifications after stop_offer_event on the server.

### Expected behaviour

server: cycle event can stop notify.
client: stopped receiving notifications

**Comment by zhaoxin39913** [comment]
Yes, I come across the same issue, even after stopping service the event still can not be stopped.

**Comment by hjkcst** [comment]
Hello, everyone

Does anyone know how to promote the modification of this problem?

thanks,
hjkcst

**Comment by zhaoxin39913** [comment]
Hi @hjkcst ，

I think you can follow the community process to contribute your change to related git branch.

**Comment by hjkcst** [comment]
HI, everyone

Has you known this problem?
![image](https://github.com/COVESA/vsomeip/assets/86393732/51bee93e-870f-4fbd-baab-e430bcf22c99)

**Comment by BrunoLSilvaCTW** [comment]
Hi @hjkcst,

You need to create a fork first. See this [page](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project) for instructions on how to do it.

**Comment by duartenfonseca** [comment]
hi @hjkcst @007herelsp @zhaoxin39913  i am trying to reproduce your problem. for that i used the following network_test: [cyclic_event_test]( https://github.com/COVESA/vsomeip/tree/master/test/network_tests/cyclic_event_tests).
i did a change to add a stop_offer_event after the notify of the service. 

i wasn't able to see the same problem. can you point out some code for the example, or why in my case it didn't work?
